cmd/boulder-va/main.go

```diff
@@ -26,7 +26,8 @@ type Config struct {
 		// before giving up. May be short-circuited by deadlines. A zero value
 		// will be turned into 1.
 		DNSTries                  int
-		DNSResolver               string `validate:"required"`
+		DNSResolver               string           `validate:"required_without=DNSProvider"`
+		DNSProvider               *cmd.DNSProvider `validate:"required_without=DNSResolver"`
```

Comment:
I think I prefer required_without over excluded_with, because required makes more sense to my mind. Are they not functionally the same, just inverse statements?

To your latter suggestion, config validation is not yet enabled by default so any checking we perform needs to be duplicated inside of main as well.

Reply:
`required_without` allows the possibility that _both_ are set, just like `excluded_with` allows the possibility that neither is set. We need both in order to get the logic we want, as far as I can tell.

Reply:
Oh, nice catch! All fixed up.